dags/map_reproducibility/utils/common_utils.py

```diff
@@ -213,12 +213,20 @@ def get_gcs_automation_repo(
 
 
 def configure_project_and_cluster(cluster: str, cluster_region: str):
-  set_project_command = (
-      f"gcloud config set project {PROJECT}",
-      "sudo chown -R airflow:airflow /home/airflow/composer_kube_config",
-      "gcloud container clusters get-credentials "
-      f"{cluster} --region {cluster_region}",
-  )
+  if is_dns(cluster):
+    set_project_command = (
+        f"gcloud config set project {PROJECT}",
+        "sudo chown -R airflow:airflow /home/airflow/composer_kube_config",
+        "gcloud container clusters get-credentials "
+        f"{cluster} --region {cluster_region} --dns-endpoint",
+    )
+  else:
+    set_project_command = (
+        f"gcloud config set project {PROJECT}",
+        "sudo chown -R airflow:airflow /home/airflow/composer_kube_config",
+        "gcloud container clusters get-credentials "
+        f"{cluster} --region {cluster_region}",
```

Comment:
most of the lines here are the same as in the "if" statement. Lets take them out of the "if" and add a comment for the if and command.

Reply:
instead of having the "if", one idea is to have a function for get-redential that gets is_dns as parameter and returns the right command